rust/candid/src/ser.rs

```diff
@@ -28,13 +27,12 @@ impl IDLBuilder {
             value_ser: ValueSerializer::new(),
         }
     }
-    pub fn arg<'a, T: types::CandidType>(&'a mut self, value: &T) -> Result<&'a mut Self> {
+    pub fn arg<'a, T: ?Sized + CandidTyping>(&'a mut self, value: &T) -> Result<&'a mut Self> {
```

Comment:
I was expecting that we can get rid of `types::internal::env_clear()` on line 24. If not, what's the point of this PR?